src/tests/suite/daos_obj.c

```diff
@@ -1458,24 +1458,51 @@ punch_simple_internal(void **state, daos_obj_id_t oid)
 			    DAOS_TX_NONE, &req);
 
 
-	/* TODO Record enumeration still under development, use lookup. */
-	print_message("Lookup non-punched records:\n");
+	/**
+	 * Lookup and enumerate records for both punched and non-punched
+	 * entries.
+	 */
+	print_message("Lookup & enumerate non-punched records:\n");
 	for (i = 0; i < PUNCH_NUM_KEYS; i++) {
 		if (i % 2 == 0)
 			num_rec_exts = PUNCH_SCM_NUM_EXTS;
 		else
 			num_rec_exts = PUNCH_NVME_NUM_EXTS;
 		size = PUNCH_IOD_SIZE * num_rec_exts;
 		D_ALLOC(rec_fetch, size);
-		print_message("\tlookup:dkey:%s, akey:'akey'\n", dkeys[i]);
+
+		/* Lookup all non-punched records */
 		lookup_single_with_rxnr(dkeys[i], "akey", /*idx*/0, rec_fetch,
 					PUNCH_IOD_SIZE, size, DAOS_TX_NONE,
 					&req);
 		assert_memory_equal(rec_fetch, data_buf, size);
+		/* Enumerate all non-punched records */
+		memset(&anchor_out, 0, sizeof(anchor_out));
+		enum_num = 2;
```

Comment:
Not really introduced by your code but I'd prefer we use named constants (e.g. #define ENUM_NUM 2) rather than copying 2 everywhere which makes it hard to read and to modify correctly when debugging.

.e.g. 
enum_num = ENUM_NUM;

and struct definitions should use [ENUM_NUM] as well (or other named constant)